src/api/traits.h

```diff
@@ -0,0 +1,50 @@
+#ifndef HERMES_TRAITS_H
+#define HERMES_TRAITS_H
+
+#include <unordered_map>
+
+#include "hermes_types.h"
+
+namespace hermes {
+namespace api {
+struct BlobInfo {
+  std::string bucket_name;
+  std::string blob_name;
+};
+typedef BlobInfo TraitInput;
+// typedef void* TraitCallback;
+
+struct Trait;
+typedef std::function<void(TraitInput &, Trait *)> TraitCallback;
+// typedef void(*TraitCallback)(TraitInput &, Trait *);
+struct Trait {
+  TraitID id;
+  TraitIdArray conflict_traits;
+  TraitType type;
+  TraitCallback onAttachFn;
+  TraitCallback onDetachFn;
+  TraitCallback onLinkFn;
+  TraitCallback onUnlinkFn;
+  Trait(TraitID id, TraitIdArray conflict_traits, TraitType type);
+};
+
+#define FILE_TRAIT 10
```

Comment:
Our convention is to prefix preprocessor symbols with `HERMES_`.

Reply:
Done.